var/spack/repos/builtin/packages/paraver/package.py

```diff
@@ -7,24 +7,69 @@
 
     # NOTE: Paraver provides only latest version for download.
     #       Don't keep/add older versions.
-    version('4.6.3', sha256='ac6025eec5419e1060967eab71dfd123e585be5b5f3ac3241085895dbeca255a')
-    version('4.6.2', sha256='74b85bf9e6570001d372b376b58643526e349b1d2f1e7633ca38bb0800ecf929')
+    version('4.9.2',     sha256='83289584040bcedf8cab1b2ae3545191c8bdef0e11ab62b06e54cbf111f2127a')
+    version('4.9.1',     sha256='e89fdf563d1fc73ed0018cf0e25b458b6617ec33325ed3fdbf06397c556f3a8e')
+    version('4.9.0',     sha256='1f9964d7987032d01a354327845bf53ae369be5d8acf7d4e85bec81699a6ddf6')
+    version('4.8.2',     sha256='0d22ec362e4798d7ed21b15b607859e9bda9579e3d5b23af3812c1e32ccc192d')
+    version('4.8.1',     sha256='d03b04254bd3015d61374b95aeda6888f593be67286c5268849623baa2ae6e2e')
+    version('4.8.0',     sha256='780af8fff7cb40d1325260fb9f79210f6676f07357bc9b95b1b838862f2d1e5b')
+    version('4.7.2',     sha256='90107797d6af6fc3ebd9505445bb518d673edecbe5d08d1b7af01695d53241ae')
+    version('4.7.1',     sha256='8cbec0c5e0f8a849820f6682cbb0920ea234bb7f20d1483e38ea5d0b0ee045cd')
+    version('4.7.0',     sha256='81e02bcc1853455b13435172a4336ba85ba05020887d322c9678c97def03d76f')
+    version('4.6.4.rc1', sha256='f7555c91c967f9a7a92ea533100326a5151e88f159483dc0d2bd1ae8c3caddf8')
+    version('4.6.3',     sha256='ac6025eec5419e1060967eab71dfd123e585be5b5f3ac3241085895dbeca255a')
+    version('4.6.2',     sha256='74b85bf9e6570001d372b376b58643526e349b1d2f1e7633ca38bb0800ecf929')
 
-    depends_on("boost")
-    # depends_on("extrae")
-    depends_on("wxwidgets")
-    depends_on("wxpropgrid")
+    depends_on('boost@1.36: +serialization')
+    depends_on('wxwidgets@2.8:')  # NOTE: using external for this one is usually simpler
+    depends_on('wxpropgrid@1.4:')
+    depends_on('libxml2')
+    depends_on('zlib')
 
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    @when('@4.6.3:')
+    def configure_args(self):
+        spec = self.spec
+        args = []
+
+        args.append('--prefix=%s' % self.prefix)
+        if spec.satisfies('@4.6.3'):
+            args.append('--with-paraver=%s' % self.prefix)
+            args.append('--with-ptools-common-files=%s' % self.prefix)
+        args.append('--with-boost=%s' % spec['boost'].prefix)
+        args.append('--with-wx-config=%s/wx-config' % spec['wxwidgets'].prefix.bin)
+        if spec['wxwidgets'].satisfies('@:2'):
+            args.append('--with-wxpropgrid=%s' % spec['wxpropgrid'].prefix)
+
+        return args
+
+    @when('@4.6.2')  # do not have global build system
+    def autoreconf(self, spec, prefix):
+        pass
+
+    @when('@4.6.2')  # do not have global build system
+    def configure(self, spec, prefix):
+        pass
+
+    # use make install directly as expected by Paraver
+    def build(self, spec, prefix):
```

Comment:
Missing `@when`

Reply:
I believe it is a always the case actually.

Reply:
Does it cause a build failure if you run `make && make install`? Most packages can be built with just `make install` but it's nice to separate them into different steps in case you want to inspect the compiled package without installing.

Reply:
I don't know, I just followed the IMPORTANT note the developers put in the README of the tar.gz :

![image](https://user-images.githubusercontent.com/1702404/121955901-ddea9b00-cd36-11eb-8f7c-b45f845470f5.png)